src/main/java/logic/parsers/DukeParser.java

```diff
@@ -0,0 +1,94 @@
+package logic.parsers;
+
+import logic.commands.Command;
+import utils.DukeException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import java.lang.reflect.Method;
+
+/**
+ * deals with making sense of the user command
+ */
+public class DukeParser {
+    //@@author chenyuheng
+    /**
+     * <p>Parse a command line String to a Commands.Command object.</p>
+     *
+     * @param line the input command line String
+     * @return the new Commands.Command object
+     * @throws DukeException if the format of command cannot be parsed
+     */
+    public static Command commandLine(String line) throws DukeException {
+        String[] splites = line.trim().split("\\s+", 2);
+        if (splites.length == 1) {
+            String[] newSplites = new String[2];
+            newSplites[0] = splites[0];
+            newSplites[1] = "";
+            splites = newSplites;
+        }
+        splites[0] = splites[0].trim().toUpperCase();
+        String[] dict = {
+            "ADD", "LIST", "DONE", "BYE", "DELETE", "FIND", "RECURRING", "SNOOZE",
+            "SCHEDULE", "CHECK", "LINK", "UNLINK", "REMOVE", "HELP"
+        };
+        splites[0] = SpellingErrorCorrector.commandCorrector(dict, splites[0]);
+        Command temp = null;
+        try {
+            Method[] parse = new Method[dict.length];
```

Comment:
This array of Method objects is similary to the array of function pointers in C.